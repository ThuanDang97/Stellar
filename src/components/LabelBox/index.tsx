import React, { memo } from 'react'
import { useTheme } from 'styled-components'

// Props type
import { LabelBoxProps } from '@self-types/components/LabelBox.props'

// Components
import { ViewStyled, TextStyled } from '@components/styled-components'

const LabelBox = ({
  title,
  isActive = false,
  handleOnClick,
  padding,
}: LabelBoxProps) => {
  const theme = useTheme()

  return (
    <ViewStyled
      onClick={handleOnClick}
      data-testid="label-box"
      bgColor={isActive ? theme.colors.frenchRose : theme.colors.silverLight}
      padding={padding}
      textAlign="center"
      cursor="pointer"
    >
      <TextStyled
        fontSize={theme.typography.fontSize.base}
        color={isActive ? theme.colors.white : theme.colors.shark}
      >
        {title}
      </TextStyled>
    </ViewStyled>
  )
}

export default memo(LabelBox)
