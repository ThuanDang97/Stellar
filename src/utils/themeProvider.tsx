import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
