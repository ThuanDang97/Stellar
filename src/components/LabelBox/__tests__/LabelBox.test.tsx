// Library
import { cleanup, screen, fireEvent } from '@testing-library/react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Props type
import { LabelBoxProps } from '@self-types/components/LabelBox.props'

// Components
import LabelBox from '@components/LabelBox'

// Define props

const props: LabelBoxProps = {
  title: 'Label',
  handleOnClick: jest.fn(),
}

describe('LabelBox Component', () => {
  afterEach(cleanup)
  const component = renderWithTheme(<LabelBox {...props} />)

  it('should render LabelBox component ', () => {
    const { asFragment } = component

    expect(asFragment()).toMatchSnapshot()
  })

  it('should be renders LabelBox active snapshot', () => {
    const newProps: LabelBoxProps = {
      ...props,
      isActive: true,
    }

    const { asFragment } = renderWithTheme(<LabelBox {...newProps} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call handleOnClick when click LabelBox', async () => {
    renderWithTheme(<LabelBox data-testid="LabelBox" {...props} />)

    fireEvent.click(await screen.findByTestId('label-box'))

    expect(props.handleOnClick).toHaveBeenCalled()
  })
})
