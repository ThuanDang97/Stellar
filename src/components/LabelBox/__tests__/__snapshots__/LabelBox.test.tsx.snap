// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelBox Component should be renders LabelBox active snapshot 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 dyOIxK"
    cursor="pointer"
    data-testid="label-box"
  >
    <span
      class="TextStyled-sc-9xia41-0 uSkVX"
      color="#FFF"
      font-size="14"
    >
      Label
    </span>
  </div>
</DocumentFragment>
`;

exports[`LabelBox Component should render LabelBox component  1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 cqCwxi"
    cursor="pointer"
    data-testid="label-box"
  >
    <span
      class="TextStyled-sc-9xia41-0 dItzST"
      color="#242A30"
      font-size="14"
    >
      Label
    </span>
  </div>
</DocumentFragment>
`;
